### PR TITLE
Fix: storing IPFS folder CID in OG Founders NFT contract

### PR DESCRIPTION
**Pull request by cybervoid0**
This pull request fixes the OG Founders NFT contract to store the IPFS folder CID in the contract. It adds a new event `IpfsFolderChanged` to track changes in the IPFS folder. The `initialize` function is updated to include the `maxSupply` and `ipfsFolderCid` parameters. The `setIpfsFolder` function is added to allow an admin to update the IPFS folder and maximum supply of tokens.

### CODE REVIEW

File: contracts/NFT/OGFoundersRootstockCollective.sol [hunk 1]
```diff
@@ -24,19 +25,37 @@ contract OGFoundersRootstockCollective is ERC721NonTransferrableUpgradable {
   address public stRIF;
   uint256 public firstProposalDate;
   // Counter for the total number of minted tokens
-  uint8 private _totalMinted;
+  uint256 private _totalMinted;
```
Comment (TravellerOnTheRun) [inline contracts/NFT/OGFoundersRootstockCollective.sol:28]: Why have you changed for uint256? What's the purpose? It will never get past 256, cause the total is 150 and for this collection that's it.

File: ignition/modules/OGFoundersModule.ts [hunk 2]
```diff
@@ -8,13 +8,20 @@ export const OGFoundersModule = buildModule('OGFounders', m => {
...
   // deploy proxy
   const proxy = m.contract('ERC1967Proxy', [
     implementation,
-    m.encodeFunctionCall(implementation, 'initialize', [deployer, stRIFAddress, firstProposalDate], {
-      id: 'Proxy',
-    }),
+    m.encodeFunctionCall(
+      implementation,
+      'initialize',
+      [deployer, stRIFAddress, firstProposalDate, maxSupply, ipfsFolderCid],
```
Comment (TravellerOnTheRun) [inline ignition/modules/OGFoundersModule.ts:20]: Why are you setting maxSupply from the deployment? What's the purpose of this change?

**Review by TravellerOnTheRun — DISMISSED** [review]
Unclear changes made which to me look absolutely unnecessary considering the fact that each collection will have its own contract and this one will not have more than 150 tokens

File: contracts/NFT/OGFoundersRootstockCollective.sol (hunk 1 above)
Comment (jessgusclark) [reply contracts/NFT/OGFoundersRootstockCollective.sol:28]: I was under the impression that if you use uint8 in isolation the compiler will automatically convert it to uint256. The way to prevent this would be to have enough uint8s to cover the entire slot. That being said, not the time to debate this.

Comment (cybervoid0) [reply contracts/NFT/OGFoundersRootstockCollective.sol:28]: Because now it’s possible to adjust the token quantity using the setIpfFOlder function. The gas savings here are minimal and matter far less than the potential overflow that could occur when adding new tokens, where the uint8 array may not be large enough to accommodate everything.

File: ignition/modules/OGFoundersModule.ts (hunk 2 above)
Comment (cybervoid0) [reply ignition/modules/OGFoundersModule.ts:20]: Because now the token quantity depends on the number of JSON files in the IPFS folder, and it should be set dynamically either in the initializer or in the setIpfsFolder function.

Comment (shenshin) [reply ignition/modules/OGFoundersModule.ts:20]: Can you guarantee that tomorrow Toby wont come and ask to replace some images or to add some? It already happened in the past and will happen again. So it must be possible to change te number of tokens in the contract

**Comment by cybervoid0** [comment]
Can you guarantee that tomorrow Toby wont come and ask to replace some images or to add some? It already happened in the past and will happen again. So it must be possible to change te number of tokens in the contract

**Review by fk-212 — APPROVED** [review]

**Review by jessgusclark — APPROVED** [review]
Thank you for the quick work!
